lib/daru/io/sql_data_source.rb

```diff
@@ -78,10 +97,19 @@ def init_adapter(db, query)
           DbiAdapter.new(db, query)
         when ActiveRecord::ConnectionAdapters::AbstractAdapter
           ActiveRecordConnectionAdapter.new(db, query)
+        when String
```

Comment:
Thank you for the feedback. I think that I will instead implement a method that looks at the db argument and does something intelligent with it before passing it on to the case statement. This way we have a hook for implementing connection strings or files etc.